packages/remix-react/components.tsx

```diff
@@ -975,32 +976,16 @@ function dedupe(array: any[]) {
   return [...new Set(array)];
 }
 
-// TODO: Can this be re-exported from RR?
-export interface RouteMatch {
-  /**
-   * The id of the matched route
-   */
-  id: string;
-  /**
-   * The pathname of the matched route
-   */
-  pathname: string;
-  /**
-   * The dynamic parameters of the matched route
-   *
-   * @see https://remix.run/file-conventions/routes-files#dynamic-route-parameters
-   */
-  params: Params<string>;
-  /**
-   * Any route data associated with the matched route
-   */
-  data: any;
-  /**
-   * The exported `handle` object of the matched route.
-   *
-   * @see https://remix.run/route/handle
-   */
-  handle: undefined | { [key: string]: any };
+export interface UIMatch<D = AppData> extends UIMatchRR<SerializeFrom<D>> {}
```

Comment:
I kept this as an interface to match what it was - is there a functional reason to prefer `type` here?

Reply:
Functionally they're _kinda_ equivalent, but:

- Interfaces can't express unions, mapped types, or conditional types. Type aliases can express any type.
- Interfaces with the same name in the same scope merge their declarations, leading to unexpected bugs
  Since this is a generic and it's not meant to be overridden, I think it's better to be safe than sorry and not make any (no pun intended 😅) unwanted side-effects possible
- Type aliases have an implicit index signature of `Record<PropertyKey, unknown>`